x-pack/legacy/plugins/epm/server/lib/kibana/index_pattern/install.ts

```diff
@@ -5,23 +5,25 @@
  */
 
 import { SavedObjectsClientContract } from 'kibana/server';
-import { KibanaAssetType } from '../../../../common/types';
+import { KibanaAssetType, DatasetType } from '../../../../common/types';
 import { RegistryPackage, Dataset } from '../../../../common/types';
-import { loadFieldsFromYaml } from '../../elasticsearch/template/install';
 import * as Registry from '../../../registry';
+import { loadFieldsFromYaml, Fields, Field } from '../../fields/field';
 
-export interface Field {
+export interface IndexPatternField {
```

Comment:
It'd be great to make this type more specific (e.g. `Pick<Field, 'count', 'index', ...>`) and representative of the logic in `transformField` (`IndexPatternField = BinaryField | ObjectField | ScriptField | ...`) but that can wait for a follow up PR/ticket.

Reply:
I'm not sure the `Pick` would make sense because if you look at the types, they are optional in `Field` and not optional in `IndexPatternField`.

Reply:
That was just to illustrate that we could express one in terms of the other (as is done in the transform function). To marks those fields as required/non-nullable we could wrap `Pick` with `Required` like 

```typescript
Required<Pick<Field, 'x' | 'y' | 'z'>>
```

Reply:
Will create another issue to think on it.  I think it's a bit more difficult to read but perhaps its best to connect the types in some way.